graspy/subgraph/sg.py

```diff
@@ -0,0 +1,181 @@
+        """
+        Fit the signal-subgraph estimator according to the constraints given.
+
+        Parameters
+        ----------
+        constraints: int or vector
+            The constraints that will be imposed onto the estimated signal-subgraph.
+            
+            If *constraints* is an int, *constraints* is the number of edges in the signal-subgraph.
+            If *constraints* is a vector, the first element of *constraints* is the number of edges in the signal-subgraph, and the second element of *constraints* is the number of vertices that the signal-subgraph must be incident to.
+        
+        Returns
+        -------
+        self: returns an instance of self
+        """
+        self.__construct_contingency()
+        verts = np.shape(self.graphs)[0]
+        sigmat = np.array(
+            [
+                [fisher_exact(self.contmat[i, j, :, :])[1] for j in range(verts)]
+                for i in range(verts)
+            ]
+        )
+
+        if isinstance(constraints, (int)):  # incoherent
+            nedges = constraints
+            sigsub = np.dstack(
+                np.unravel_index(np.argsort(sigmat.ravel()), np.shape(sigmat))
+            )
+            sigsub = sigsub[0, :nedges, :]
+            sigsub = tuple(np.transpose(sigsub))
+
+        elif len(constraints) == 2:  # coherent
+            nedges = constraints[0]
+            nverts = constraints[1]
+
+            wset = np.unique(sigmat, axis=None)
+            wcounter = 0
+            wconv = 0
+
+            while wconv == 0:
+                w = wset[wcounter]
+                blank = sigmat
+                blank = blank > w
+
+                score = 2 * verts - (np.sum(blank, axis=1) + np.sum(blank, axis=0))
+                vscore = np.sort(score)[::-1]
+                vstars = np.argsort(score)[::-1]
+
+                if (vscore[:nverts].sum()) >= nedges:
+                    blank = np.ones(np.shape(sigmat))
+                    nstars = np.amin([len(vscore[vscore > 0]), nverts])
+                    vstars = vstars[:nstars]
+
+                    blank[vstars, :] = sigmat[vstars, :]
+                    blank[:, vstars] = sigmat[:, vstars]
+
+                    indsp = np.dstack(
+                        np.unravel_index(np.argsort(blank.ravel()), np.shape(blank))
+                    )
+                    sigsub = indsp[0, :nedges, :]
+                    sigsub = tuple(np.transpose(sigsub))
+                    wconv = 1
+                else:
+                    wcounter = wcounter + 1
+                    if wcounter > len(wset):
+                        sigsub = []
+                        wconv = 1
+        else:
+            msg = "Input constraints must be an int for the incoherent signal-subgraph estimator, or a vector of length 2 for the coherent subgraph estimator."
+            raise TypeError(msg)
+        self.sigsub = sigsub
+        return self
+
+    def fit_transform(self, constraints, return_mask=False):
```

Comment:
Same thing. Add `X` and `y` as inputs here. I think you could just compute mask and save as class attribute called `mask_` and not having this option.